### ISSUE TITLE
Mac Install Failure

### ISSUE THREAD

**Issue by byronmansfield** [input-output-hk/daedalus#2567]
I think I found a bug. Forgive me if it's already been posted. I just downloaded `daedalus-4.0.5-mainnet-17596.pkg` directly from the website. I am running macOS 10.15.7. Here is the `/var/log/install.log` from the install.

```
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: @(#)PROGRAM:Install  PROJECT:Install-1000
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: @(#)PROGRAM:Installer  PROJECT:Installer-965.1
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Hardware: MacBookPro15,2 @ 2.80 GHz (x 8), 16384 MB RAM
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Running OS Build: Mac OS X 10.15.7 (19H2)
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Env: USER=bmansfield
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Env: COMMAND_MODE=unix2003
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Env: DISPLAY=/private/tmp/com.apple.launchd.MN6cE1ZZDg/org.macosforge.xquartz:0
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Env: PATH=/usr/bin:/bin:/usr/sbin:/sbin
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Env: LOGNAME=bmansfield
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Env: SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.X82xJhpvgj/Listeners
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Env: HOME=/Users/bmansfield
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Env: SHELL=/bin/zsh
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Env: TMPDIR=/var/folders/qd/nm41qmvn1hq34rppc_1h0dy40000gn/T/
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Env: __CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Env: XPC_SERVICE_NAME=com.apple.xpc.launchd.oneshot.0x1000004b.Installer
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Env: XPC_FLAGS=0x0
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]:   Installation Log
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Opened from: /Users/bmansfield/Downloads/daedalus-4.0.5-mainnet-17596.pkg
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Package Authoring Error: <background_scaling> has an unsupported MIME type: X-NSObject/NSNumber
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Package Authoring Error: <background_alignment> has an unsupported MIME type: X-NSObject/NSNumber
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Package Authoring Error: <layout-direction> has an unsupported MIME type: X-NSObject/NSNumber
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Failed to load specified background image
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Product archive /Users/bmansfield/Downloads/daedalus-4.0.5-mainnet-17596.pkg trustLevel=350
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: External component packages (1) trustLevel=350
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Could not load resource readme: (null)
2021-05-16 16:51:05-07 Byrons-MacBook-Pro-2 Installer[22631]: Could not load resource license: (null)
2021-05-16 16:51:09-07 Byrons-MacBook-Pro-2 Installer[22631]: ================================================================================
2021-05-16 16:51:09-07 Byrons-MacBook-Pro-2 Installer[22631]: User picked Standard Install
2021-05-16 16:51:09-07 Byrons-MacBook-Pro-2 Installer[22631]: Choices selected for installation:
2021-05-16 16:51:09-07 Byrons-MacBook-Pro-2 Installer[22631]:   Install: ""
2021-05-16 16:51:09-07 Byrons-MacBook-Pro-2 Installer[22631]:   Install: "(null)"
2021-05-16 16:51:09-07 Byrons-MacBook-Pro-2 Installer[22631]:           daedalus-4.0.5-mainnet-17596.pkg#daedalus-4.0.5-mainnet-17596.pkg : org.Daedalusmainnet.pkg : 0
2021-05-16 16:51:09-07 Byrons-MacBook-Pro-2 Installer[22631]: ================================================================================
2021-05-16 16:51:09-07 Byrons-MacBook-Pro-2 Installer[22631]: It took 0.00 seconds to summarize the package selections.
2021-05-16 16:51:09-07 Byrons-MacBook-Pro-2 Installer[22631]: -[IFDInstallController(Private) _buildInstallPlanReturningError:]: location = file://localhost
2021-05-16 16:51:09-07 Byrons-MacBook-Pro-2 Installer[22631]: -[IFDInstallController(Private) _buildInstallPlanReturningError:]: file://localhost/Users/bmansfield/Downloads/daedalus-4.0.5-mainnet-17596.pkg#daedalus-4.0.5-mainnet-17596.pkg
2021-05-16 16:51:09-07 Byrons-MacBook-Pro-2 Installer[22631]: Set authorization level to root for session
2021-05-16 16:51:15-07 Byrons-MacBook-Pro-2 Installer[22631]: Administrator authorization granted.
2021-05-16 16:51:15-07 Byrons-MacBook-Pro-2 Installer[22631]: Will use PK session
2021-05-16 16:51:15-07 Byrons-MacBook-Pro-2 Installer[22631]: Using authorization level of root for IFPKInstallElement
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 Installer[22631]: Starting installation:
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 Installer[22631]: Configuring volume "Macintosh HD"
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 Installer[22631]: Preparing disk for local booted install.
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 Installer[22631]: Free space on "Macintosh HD": 624.97 GB (624972210176 bytes).
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 Installer[22631]: Create temporary directory "/var/folders/qd/nm41qmvn1hq34rppc_1h0dy40000gn/T//Install.22631M8Z623"
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 Installer[22631]: IFPKInstallElement (1 packages)
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 Installer[22631]: Current Path: /System/Library/CoreServices/Installer.app/Contents/MacOS/Installer
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: Adding client PKInstallDaemonClient pid=22631, uid=501 (/System/Library/CoreServices/Installer.app/Contents/MacOS/Installer)
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 Installer[22631]: PackageKit: Enqueuing install with framework-specified quality of service (utility)
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: ----- Begin install -----
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: request=PKInstallRequest <1 packages, destination=/>
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: packages=(
            "PKLeopardPackage <id=org.Daedalusmainnet.pkg, version=0, url=file:///Users/bmansfield/Downloads/daedalus-4.0.5-mainnet-17596.pkg#daedalus-4.0.5-mainnet-17596.pkg>"
        )
2021-05-16 16:51:16-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: Set reponsibility for install to 22631
2021-05-16 16:51:19-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: Extracting file:///Users/bmansfield/Downloads/daedalus-4.0.5-mainnet-17596.pkg#daedalus-4.0.5-mainnet-17596.pkg (destination=/Library/InstallerSandboxes/.PKInstallSandboxManager/83EFF773-5062-429A-A2D3-E1C0FC17B254.activeSandbox/Root/Applications, uid=0)
2021-05-16 16:51:25-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: prevent user idle system sleep
2021-05-16 16:51:25-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: suspending backupd
2021-05-16 16:51:25-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: Using trashcan path /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/PKInstallSandboxTrash/83EFF773-5062-429A-A2D3-E1C0FC17B254.sandboxTrash for sandbox /Library/InstallerSandboxes/.PKInstallSandboxManager/83EFF773-5062-429A-A2D3-E1C0FC17B254.activeSandbox
2021-05-16 16:51:25-07 Byrons-MacBook-Pro-2 install_monitor[26445]: Temporarily excluding: /Applications, /Library, /System, /bin, /private, /sbin, /usr
2021-05-16 16:51:27-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: Shoving /Library/InstallerSandboxes/.PKInstallSandboxManager/83EFF773-5062-429A-A2D3-E1C0FC17B254.activeSandbox/Root (1 items) to /
2021-05-16 16:51:27-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit (package_script_service): Preparing to execute script "./postinstall" in /private/tmp/PKInstallSandbox.LyA7df/Scripts/org.Daedalusmainnet.pkg.Wj0Yi0
2021-05-16 16:51:27-07 Byrons-MacBook-Pro-2 package_script_service[86605]: PackageKit: Executing script "postinstall" in /tmp/PKInstallSandbox.LyA7df/Scripts/org.Daedalusmainnet.pkg.Wj0Yi0
2021-05-16 16:51:27-07 Byrons-MacBook-Pro-2 package_script_service[86605]: Set responsibility to pid: 22631, responsible_path: /System/Library/CoreServices/Installer.app/Contents/MacOS/Installer
2021-05-16 16:51:27-07 Byrons-MacBook-Pro-2 package_script_service[86605]: ./postinstall:   File "./dockutil", line 106
2021-05-16 16:51:27-07 Byrons-MacBook-Pro-2 package_script_service[86605]: ./postinstall:     """ % dict(progname = os.path.basename(sys.argv[0]))
2021-05-16 16:51:27-07 Byrons-MacBook-Pro-2 package_script_service[86605]: ./postinstall:       ^
2021-05-16 16:51:27-07 Byrons-MacBook-Pro-2 package_script_service[86605]: ./postinstall: SyntaxError: invalid syntax
2021-05-16 16:51:27-07 Byrons-MacBook-Pro-2 package_script_service[86605]: Responsibility set back to self.
2021-05-16 16:51:27-07 Byrons-MacBook-Pro-2 install_monitor[26445]: Re-included: /Applications, /Library, /System, /bin, /private, /sbin, /usr
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: releasing backupd
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: allow user idle system sleep
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: Cleared responsibility for install from 22631.
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: Cleared permissions on Installer.app
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: Install Failed: Error Domain=PKInstallErrorDomain Code=112 "An error occurred while running scripts from the package “daedalus-4.0.5-mainnet-17596.pkg”." UserInfo={NSFilePath=./postinstall, NSURL=file:///Users/bmansfield/Downloads/daedalus-4.0.5-mainnet-17596.pkg#daedalus-4.0.5-mainnet-17596.pkg, PKInstallPackageIdentifier=org.Daedalusmainnet.pkg, NSLocalizedDescription=An error occurred while running scripts from the package “daedalus-4.0.5-mainnet-17596.pkg”.} {
            NSFilePath = "./postinstall";
            NSLocalizedDescription = "An error occurred while running scripts from the package \U201cdaedalus-4.0.5-mainnet-17596.pkg\U201d.";
            NSURL = "file:///Users/bmansfield/Downloads/daedalus-4.0.5-mainnet-17596.pkg#daedalus-4.0.5-mainnet-17596.pkg";
            PKInstallPackageIdentifier = "org.Daedalusmainnet.pkg";
        }
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: Running idle tasks
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 Installer[22631]: install:didFailWithError:Error Domain=PKInstallErrorDomain Code=112 "An error occurred while running scripts from the package “daedalus-4.0.5-mainnet-17596.pkg”." UserInfo={NSFilePath=./postinstall, NSURL=file:///Users/bmansfield/Downloads/daedalus-4.0.5-mainnet-17596.pkg#daedalus-4.0.5-mainnet-17596.pkg, PKInstallPackageIdentifier=org.Daedalusmainnet.pkg, NSLocalizedDescription=An error occurred while running scripts from the package “daedalus-4.0.5-mainnet-17596.pkg”.}
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: Removing client PKInstallDaemonClient pid=22631, uid=501 (/System/Library/CoreServices/Installer.app/Contents/MacOS/Installer)
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 installd[78040]: PackageKit: Done with sandbox removals
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 Installer[22631]: Install failed: The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 Installer[22631]: IFDInstallController 252BA0 state = 8
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 Installer[22631]: Displaying 'Install Failed' UI.
2021-05-16 16:51:28-07 Byrons-MacBook-Pro-2 Installer[22631]: 'Install Failed' UI displayed message:'The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.'.
```

Let me know what other helpful information I can leave on this issue in order to help you troubleshoot. Thank you in advance.

**Comment by tiwong-ln2** [comment]
Looks like a Python syntax issue in printing the docstring in the directory "daedalus/installers/data/scripts/dockutil" due to some missing parentheses.

**Comment by byronmansfield** [comment]
Yes. That is what it looks like. I do software, but I've never built any desktop apps so I have no clue how this works. But I'm very curious to know how this downloaded .pkg which is downloaded and installed by many others, doesn't have this same syntax error. Is it possible it's my python installation? Either way, even if it was my python installation, wouldn't it still be a syntax error for others? I've confirmed at least one other with the same OSX version as mine, and installed this same daedalus version, without this issue. So how could this be possible? Do we have different packages downloaded? Where is this python script? If I could find it, I know enough python where I could probably fix the syntax error. Is that possible to do with a mac .pkg installer?

**Comment by nikolaglumac** [comment]
@byronmansfield may I kindly ask you to re-download the package and try to install again? https://daedaluswallet.io/en/download/
...perhaps the package was damaged...

**Comment by byronmansfield** [comment]
Of course. No problem at all. So I actually tried it twice. From two different browsers. Firefox and Chrome. Just in case some bizarre chance something from my browser could be effecting it. And I have the same result/error from both downloads.

**Comment by nikolaglumac** [comment]
@disassembler @cleverca22 can you guys please help us here?

**Comment by byronmansfield** [comment]
@nikolaglumac Is it possible to try an older version? I just don't know where I can find a download link.

**Comment by nikolaglumac** [comment]
That won't make a difference... @input-output-hk/daedalus-qa do we have OSX 10.15.7 on which we could test the installation? I would like you to try to reproduce this issue 🙏

**Comment by ManusMcCole** [comment]
Hi @nikolaglumac.I currently have Catalina 10.15.6 so im upgrading now to .7 and I will test 👍

**Comment by ManusMcCole** [comment]
I could not reproduce the issue on Catalina 10.15.7. I did 3 clean installs and all of the were succesfull @nikolaglumac @disassembler

**Comment by byronmansfield** [comment]
Thank you for testing that @ManusMcCole. Just for kicks, I downloaded it again, and attempted to re-install it again. I got the same reported results from above. What else can I do to help to diagnose the issue on my machine? Any other logs I can pull for you?

**Comment by nikolaglumac** [comment]
@byronmansfield did you ever install Daedalus in the past?

It looks like your system is detecting a syntax error in the postinstall script code on this line: https://github.com/input-output-hk/daedalus/blob/develop/installers/data/scripts/dockutil#L106
This is very unusual and I am 100% sure there is something wrong with the setup of your OSX.
Installer code is the responsibility of the DEVOPS team so we won't be able to do much without their help.
cc @cleverca22 @disassembler

**Comment by byronmansfield** [comment]
Not on this machine. I have set it up previously on another machine. Much older versions and much older OSX. I do however have ledger live setup on this machine that is having the issues. And have transferred my ADA to my ledger wallet there if that somehow has anything to do with it. I've also used the ADA Lite wallet on this machine. Neither of these are having any issues. But actually installing the Daedalus wallet on this machine. No I have not.

**Comment by byronmansfield** [comment]
@nikolaglumac So I did another reboot. I re-downloaded a fresh pkg. Then I re-ran the installer again. While I did get the same error message, I went to my applications in my dock bar to try and launch the console app to diagnose the install error, and boom. The daedalus wallet is there. I let it sync with the block chain overnight. It seems to be working now? Do you think it is safe enough to try something like staking? Is there any further logs or tests you might suggest running before considering the installation a success. To be clear, it still gives me the python syntax error.

**Comment by nikolaglumac** [comment]
Huh it is hard to say what happened... Perhaps you could try to uninstall Daedalus and then remove all associated data and restart your machine. Only after all these steps are done you should try to install it again.

**Comment by byronmansfield** [comment]
Sure. I probably won't have time until this weekend. But yes, I will try that.

**Comment by byronmansfield** [comment]
So I was actually able to do that much sooner. I deleted the Application, and any other related files that usually don't get deleted with the app. Application Support, Library, etc. Then I rebooted, re-downloaded, re-installed and still got the same error. However it did actually install. I'm sync'ing with the blockchain right now. I'm not sure what keeps causing this. But it does appear it's something with my local. I guess as long as it installs and runs, things should be fine? Anything you feel I should be cautious of or check on before I start doing transactions or staking?
